code/modules/projectiles/guns/energy.dm

```diff
@@ -128,3 +131,29 @@ var/global/list/registered_cyborg_weapons = list()
 			icon_state = "[modifystate][get_charge_ratio()]"
 		else
 			icon_state = "[initial(icon_state)][get_charge_ratio()]"
+
+
+//For removable cells.
+/obj/item/gun/energy/attack_hand(mob/user)
+	if(!user.is_holding_offhand(src)|| isnull(cell_type) || isnull(power_supply) )
+		return ..()
+	user.put_in_hands(power_supply)
+	power_supply = null
+	user.visible_message(SPAN_NOTICE("\The [user] unloads \the [src]."))
+	playsound(src,'sound/weapons/guns/interaction/smg_magout.ogg' , 50)
+	update_icon()
+
+/obj/item/gun/energy/attackby(var/obj/item/A, mob/user)
+	if(isnull(cell_type))
+		return
+	if(!user.is_holding_offhand(src))
+		to_chat(user, SPAN_NOTICE("Hold \The [src] in your hands to load it."))
+		return
+	if(istype(power_supply) )
+		to_chat(user, SPAN_NOTICE("There is already \a [power_supply] loaded."))
+
+	if(istype(A, cell_type) && do_after(user, 5, A, can_move = TRUE) && user.unEquip(A, src) )
```

Comment:
IMO conditionals should be pure, so unEquip should be called inside this block and then check `!user.isEquipped(A)`. That's not official style afaik, though, so feel free to dismiss this comment I guess

Reply:
Existing implementation generally only checks unEquip() after the other conditions afaik.

Reply:
I'm not quite sure what's meant by pure.